exercise_150507_Calculator.rb

```diff
@@ -0,0 +1,86 @@
+# File Name: 150507_Calculator_Exercise.rb
+# Developer: catchingash
+
+# Establish synonyms for math functions. NOTE: All text in the arrays needs to be lowercase.
+subtract = ["substraction", "sub", "subtract", "-"]
+add = ["addition", "add", "+"]
+multiply = ["multiplication", "multiply", "x", "*"]
+divide = ["division", "div", "/"]
+exponent = ["exponential", "exponents", "exponent", "exp", "power", "^", "**"]
+sqrt = ["squareroot", "square root", "sqrt", "^1/2", "\u221A"]
+mod = ["remainders", "remainder", "modulo", "modulus", "%"]
+operations = subtract + add + multiply + divide + exponent + sqrt + mod
+UNACCEPTED_OPERATION_MEMO = "I'm sorry. I'm confused. I know it's probably obvious to a human, but could you try again and dumb it down for me? (You'll need to restart me. Sorry!)"
+UNEXPECTED_ERROR_MEMO = "I'm not sure what happened. I unencountered an unexpected error and had to close. I'm sorry!"
+
+# Get user input: operation selection
+username = `id -u -n`.chomp
+puts "Hello, #{username}. Can I call you that? I hope it's okay. Anyway, let's do some math!\nWhat kind of math would you like to do? [If you need a hint, type 'HELP'.]"
+operator = gets.chomp.downcase.strip
+
+while operator == "help"
+  puts "Thanks for asking. I do lots of math! Your options are:\naddition, subtraction, multiplication, division, exponents, square root, remainder."
+  operator = gets.chomp.downcase.strip
+end
+
+abort(UNACCEPTED_OPERATION_MEMO) unless operations.include?(operator)
+
+# Get user input: number selection
+puts "Exciting! Pick the first number."
+num1 = gets.chomp.delete(",")
```

Comment:
I sent my calculator to a friend, who promptly put in "5,000" as one of the numbers. "5,000" was treated as though it was "5" instead. I added .delete(",") so it would be treated the same as "5000".